src/ArcxAnalyticsSdk.ts

```diff
@@ -148,9 +158,14 @@ export class ArcxAnalyticsSdk {
 
   private _handleAccountDisconnected() {
     if (!this.currentChainId || !this.currentConnectedAccount) {
-      throw new Error(
-        'ArcxAnalyticsSdk::_handleAccountDisconnected: previousChainId or previousConnectedAccount is not set',
-      )
+      /**
+       * It is possible that this function has already been called once and the cached values
+       * have been cleared. This can happen in the follwoing scenario:
+       * 1. Initialize ArcxAnalyticsProvider with the default config (sets MM as the initial provider)
+       * 2. Connect WalletConnect
+       * 3. Disconnect
+       */
+      return
```

Comment:
Weird edge-case. For some reason, the disconnect event is being emitted 3 times when disconnecting wallet-connect and I'm not sure why. I spend a considerable time on it and decided to move on. This is not the case with metamask.